src/collector/styles/index.ts

```diff
@@ -1,26 +1,59 @@
-import { iif, Observable } from 'rxjs';
-import { map, flatMap, toArray, tap, take } from 'rxjs/operators';
-import { config } from '../config';
-import { scan } from '../../scanner';
-import { getFile, IFile } from '../file';
-import { parseSass, IStyleFile } from './parser';
-import { printParsedFileLog } from '../log';
-import { createDebugger, createLogger } from '../../logger';
-import { environment } from '../../environment';
+import {iif, Observable, from} from 'rxjs';
+import {map, flatMap, toArray, tap, take, mergeMap, scan as scanRx, filter} from 'rxjs/operators';
+import {config} from '../config';
+import {IScanSettings, scan} from '../../scanner';
+import {getFile, IFile} from '../file';
+import {parseSass, IStyleFile} from './parser';
+import {printParsedFileLog} from '../log';
+import {createDebugger, createLogger} from '../../logger';
+import {environment} from '../../environment';
+
+type TMergeMapResult = [string, IStyleFile[]];
+export interface IStyleFilesResult { [key: string]: IStyleFile[] }
 
 const debugFile = createDebugger<IFile>('Collecting style', 'name');
-const logCollection = createLogger<IStyleFile[]>('Styles:', 'length');
-const scanForFiles = (): Observable<string> => scan(config.settings.core.scan.styles)
+const logCollection = (level: string) => createLogger<IStyleFile[]>(`${level} Styles:`, 'length');
+const scanForFiles = (configSettings: IScanSettings) => (): Observable<string> => scan(configSettings);
+const scanForFilesCollection = [
+    {
+        scanFunction: scanForFiles(config.settings.core.scan.styles),
+        scanMessage: logCollection('Vendor'),
+        scanLevel: 'vendor'
+    },
+    {
+        scanFunction: scanForFiles(config.settings.project.scan.styles),
+        scanMessage: logCollection('Project'),
+        scanLevel: 'project'
+    }
+];
+
+const restrictedScanForFilesCollection = iif(
+    () => environment.isAllowedLevel,
+    from(scanForFilesCollection).pipe(filter(item => item.scanLevel === environment.levelRestriction)),
+    from(scanForFilesCollection)
+);
 
-export const getObservable = (): Observable<IStyleFile[]> => iif(
+const limitedScanForFilesCollection = (scanForFiles) => iif(
```

Comment:
if you use only one parameter, you can write without brackets
const limitedScanForFilesCollection = scanForFiles => iif(